### DISCUSSION TITLE
[FAQ-Request] How is NewPipe different from PipePipe -- why are services deprecated?

### DISCUSSION THREAD

**Discussion by EvanCarroll** [InfinityLoop1308/PipePipe#411] [Q&A]
I'm confused about the distinction in the fork. Clearly new services were added that NewPipe didn't want to support. But does it make sense to deprecate the few services that NewPipe does support? Is the goal of this fork not to replace NewPipe but to compliment it? It seems conflicting with the text in the readme,

> NewPipe, but better - **browse the most popular streaming platforms throughout the world.**

The only three services supported and not deprecated are YouTube and two sources you added. It seems on _browsing the most popular platforms_ it's missing the most popular ones that NewPipe offers (ie, not obviously "better")

It would also be a good idea to differentiate yourself from BravenNewPipe. Ideally in my world, if the value add here is only to make the two added sources something you can access in F-droid, perhaps you could get them included in BravenNewPipe and take the job of getting it contributed upstream to F-droid. BravenNewPipe accepts sources with less _"editorial discretion"_ as NewPipe.

I would also perhaps describe the two sources you're adding in English. I have no idea even what they are or why they exist outside of YouTube. Seems to be a Japanese video game streaming thing?

**Comment by InfinityLoop1308** [comment]
1. The main advantage of PipePipe is offering richer features and a better user experience, rather than providing more services.
2. PipePipe is a hard fork of NewPipe. If you are not familiar with this concept, in simpler terms, we only used the 2022 code of NewPipe, and since then, it has been a completely independent development process. Therefore, it is neither a replacement nor a supplement. PipePipe is not aligned with NewPipe now and will not be in the future.
3. Maintaining a service is a very demanding task. I do not use these services, so I do not have the energy to make additional modifications to them. This does not mean you cannot use them - in fact, they are stable enough, have been working normally for two years, and will likely continue to work for five years or more.
What this practically means:
- I will not proactively update them or add new features to them.
- If issues arise, I will not be as proactive in fixing them as I am with other services.
   
  I admit that the term "deprecated" might sound severe here, but I cannot find a better word to describe this state.
The same issue applies to Rumble. You are not the first to invite me to join BravePipe, and I suspect you will not be the last. If you are a developer on their team, you should know that the differences between forks have become so significant that merging them would be less efficient than starting anew.

**Reply by EvanCarroll** [reply]
Do you view the value-add of the fork as being the services, or the UI? Does your hard-fork also change the extractor services or just the user interface? What value add was there in breaking compat with the extractor services? The ones currently marked as "deprecated" must have either been modified to make them work with PipePipe, or the new API is at least backwards-compat with NewPipe. If modifications were made there it would be interesting to at least get those upstreamed.

If no modifications were made to the extractor service, then it would seem trivial to get the two new extractor services here into BravenNewPipe. And the obvious question (from my perspective) would be whether or not you're adverse to that? Can they use the service you're using here less the unique interface you've created.

**Comment by InfinityLoop1308** [comment]
> I would also perhaps describe the two sources you're adding in English. I have no idea even what they are or why they exist outside of YouTube. Seems to be a Japanese video game streaming thing?

They are Japanese and Chinese platforms which are popular there.
